src/Core/Framework/App/Lifecycle/Persister/CustomFieldPersister.php

```diff
@@ -47,13 +47,27 @@ public function updateCustomFields(Manifest $manifest, string $appId, Context $c
 
     private function upsertCustomFieldSets(?CustomFields $customFields, string $appId, Context $context): void
     {
-        $existingCustomFieldSets = Uuid::fromBytesToHexList(
-            $this->connection->fetchAllKeyValue(
-                'SELECT name, id FROM custom_field_set WHERE app_id = :appId',
-                ['appId' => Uuid::fromHexToBytes($appId)]
-            )
+        $allCustomFields = $this->connection->fetchAllKeyValue(
+            'SELECT id, name FROM custom_field_set WHERE app_id = :appId',
+            ['appId' => Uuid::fromHexToBytes($appId)]
         );
 
+        $groupedByName = [];
+        foreach ($allCustomFields as $id => $name) {
+            $groupedByName[$name][] = Uuid::fromBytesToHex($id);
+        }
+
+        $existingCustomFieldSets = [];
+        foreach ($groupedByName as $name => $ids) {
```

Comment:
i didn't find a more elegant solution without the double loop, if you have any ideas how to improve please let me know

Reply:
i still would need a double loop right?